go/vt/discovery/healthcheck.go

```diff
@@ -733,10 +744,18 @@ func (hc *HealthCheckImpl) AddTablet(tablet *topodatapb.Tablet, name string) {
 		hc.mu.Unlock()
 		return
 	}
-	if _, ok := hc.addrToHealth[key]; ok {
-		hc.mu.Unlock()
-		log.Warningf("adding duplicate tablet %v for %v: %+v", name, tablet.Alias.Cell, tablet)
-		return
+	if th, ok := hc.addrToHealth[key]; ok {
+		// Something already exists at this key.
+		// If it's the same tablet, something is wrong.
+		if topoproto.TabletAliasEqual(th.latestTabletStats.Tablet.Alias, tablet.Alias) {
+			hc.mu.Unlock()
+			log.Warningf("refusing to add duplicate tablet %v for %v: %+v", name, tablet.Alias.Cell, tablet)
```

Comment:
This should eventually become an acceptable state, which will allow us to "update if needed" kind of calls. But let the warning remain for now. It will allow us to find out if this really happens in the wild.

The other future improvement on this will be to add a timestamp that tracks when a tablet acquired the address to declare the true winner. Similar to how we're solving the mastership story.